scripts/bump.sh

```diff
@@ -109,49 +120,168 @@ check_iscsc_remote () {
+	log_info 'Bumping `backend`'
+	(
+		cd backend
+		[ -z "$DRY_RUN" ] && npm version "${new_version}" --no-git-tag-version
+	)
+}
 
 # ----------------------------- Bump root and push -----------------------------
 
-log_info 'Bumping `root`'
-[ -z "$DRY_RUN" ] && { npm version "${NEW_VERSION}" --no-git-tag-version || exit 1; }
-[ -z "$DRY_RUN" ] && { git commit -m "Bump to version ${NEW_VERSION}" || exit 1; }
+bump_root () {
+	local new_version="$1"
+	log_info 'Bumping `root`'
+	[ -z "$DRY_RUN" ] && { npm version "${new_version}" --no-git-tag-version || exit 1; }
+	[ -z "$DRY_RUN" ] && { git commit -m "Bump to version ${new_version}" || exit 1; }
+}
+
+# ------------------------------- Help and usage -------------------------------
+
+usage () {
+	echo "Usage: $USAGE"
+	echo "Type -h or --help for the full help."
+	echo "See https://github.com/iScsc/iscsc.fr/wiki/Version-bump-procedure#automatic-version-bump for full documentation"
+	exit 0
+}
+
+help () {
+	echo "bump.sh:"
+	echo "     a script to easily bump the version of the iScsc website project."
+	echo "     it should guide you when you make mistakes, e.g. providing less or more than 1 argument."
+	echo ""
+	echo "Usage:"
+	echo "     $USAGE"
+	echo ""
+	echo "Options:"
+	echo "     -h/--help                shows this help."
+	echo "     -p/--patch               bump to patch version (1.2.3 --> 1.2.4)."
+	echo "     -m/--minor               bump to minor version (1.2.3 --> 1.3.0)."
+	echo "     -M/--major               bump to major version (1.2.3 --> 2.0.0)."
+	exit 0
+}
+
+# --------------------------- Version Bump Functions ---------------------------
+
+bump_major () {
+    bits=($(break_down_version $1))
+    major=${bits[0]}
+
+    echo "$((major+1)).0.0"
+}
+
+
+bump_minor () {
+    bits=($(break_down_version $1))
+    major=${bits[0]}
+    minor=${bits[1]}
+
+    echo "$major.$((minor+1)).0"
+}
+
+
+bump_patch () {
+    bits=($(break_down_version $1))
+    major=${bits[0]}
+    minor=${bits[1]}
+    patch=${bits[2]}
+
+    echo "$major.$minor.$((patch+1))"
+}
 
-log_info 'Pushing branch and bump commit'
-log_warning "pushing to \`${ISCSC_REMOTE}\` please type your passphrase/password if required:"
-PUSH_COMMAND="git push ${ISCSC_REMOTE} ${BUMP_BRANCH} v${CURRENT_VERSION}"
-[ -z "$DRY_RUN" ] && { $PUSH_COMMAND || log_error "push failed, you can push with \`${PUSH_COMMAND}\`"; }
+# -------------------------------- Main Section --------------------------------
+
+main () {
+	COMMAND=""
+	# Extract new version from arguments
+	while [[ $# -gt 0 ]]; do
+		case "$1" in
+			-h | --help ) help ;;
+			-p | --patch ) COMMAND="patch"; shift 1;;
+			-m | --minor ) COMMAND="minor"; shift 1;;
+			-M | --major ) COMMAND="major"; shift 1;;
+			-- ) shift; break ;;
+			* ) break ;;
+		esac
+	done
+
+	# Run all basic checks
+	check_pwd
+	check_arg "${NB_ARGS}"
+	check_dependencies
+	check_clean_git_working_dir
+
+	# Define version variables
+	CURRENT_VERSION=$(npm pkg get version | sed 's/"//g')
```

Comment:
maybe a little comment on this `sed` expression could be welcomed

Reply:
I could but it's a pretty basic one! juste remove every `"` encountered

Reply:
What is the goal of this operation then ?

Reply:
If you try yourself `npm pkg get version` you'll see that the ouput looks like `"0.1.1"` this is really annoying I don't why... but anyway I have to get rid of the quotes for the rest of the script

Reply:
Ok I see